storage/client_raft_test.go

```diff
@@ -1598,3 +1598,46 @@ func TestCheckInconsistent(t *testing.T) {
 		t.Fatal("didn't receive notification from VerifyChecksum() that should have panicked")
 	}
 }
+
+func TestTransferRaftLeadership(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	const numStores = 3
+	mtc := startMultiTestContext(t, numStores)
+	defer mtc.Stop()
+	// Setup replication of ramge 1 on store 0 to stores 1 and 2.
+	mtc.replicateRange(1, 1, 2)
+
+	// Write something to the DB to make range leadership and raft leadership to be located in stores[0].
+	pArgs := putArgs([]byte("a"), []byte("b"))
+	if _, err := client.SendWrapped(rg1(mtc.stores[0]), nil, &pArgs); err != nil {
```

Comment:
It shouldn't be flaky (and also this `Put` shouldn't be necessary): store 0 must have been leader (both raft and lease) for `mtc.replicateRange()` to complete. And these tests use manual clocks so we shouldn't even need to worry about something timing out and giving leadership to another store.